src/Symfony/Component/Process/Process.php

```diff
@@ -1206,14 +1184,21 @@ private function doSignal($signal, $throwException)
 
                 return false;
             }
-        }
-
-        if (true !== @proc_terminate($this->process, $signal) && '\\' !== DIRECTORY_SEPARATOR) {
-            if ($throwException) {
-                throw new RuntimeException(sprintf('Error while sending signal `%s`.', $signal));
+        } else {
+            if (!$this->enhanceSigchildCompatibility || !$this->isSigchildEnabled()) {
+                $ok = @proc_terminate($this->process, $signal);
+            } elseif (function_exists('posix_kill')) {
+                $ok = @posix_kill($this->getPid(), $signal);
+            } elseif ($ok = proc_open(sprintf('kill -%d %d', $signal, $this->getPid()), array(2 => array('pipe', 'w')), $pipes)) {
```

Comment:
what about using a synchronous call for the commandline like `exec`? When reading next line, `kill` might not have returned yet